cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java

```diff
@@ -327,13 +327,16 @@ private List<CloudVmInstanceStatus> fillCloudVmInstanceStatuses(AuthenticatedCon
         return cloudVmInstanceStatuses;
     }
 
-    private Collection<String> instanceIdsWhichAreNotInCorrectState(List<CloudInstance> vms, AmazonEc2Client amazonEC2Client, String state) {
+    private Collection<String> instanceIdsWhichAreNotInCorrectState(List<CloudInstance> vms, AmazonEc2Client amazonEC2Client, String state,
+            Set<InstanceStatus> completedStatuses) {
         Set<String> instances = vms.stream().map(CloudInstance::getInstanceId).collect(Collectors.toCollection(HashSet::new));
         DescribeInstancesResponse describeInstances = amazonEC2Client.describeInstances(
                 DescribeInstancesRequest.builder().instanceIds(instances).build());
+        Set<String> status = completedStatuses.stream().map(i -> i.toString().toLowerCase()).collect(Collectors.toUnmodifiableSet());
```

Comment:
why is this change necessary?
The completed statuses are `com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.InstanceStatus` which is our status implementation of the instances(provider statuses mapped to this) and the value what is trying to be mapped to 
 without any conversion is `software.amazon.awssdk.services.ec2.model.Instance`. 
I don't think the two types have the same value set.

Reply:
> why is this change necessary? The completed statuses are 
@biharitomi - The jira is fairly old now, and unfortunately without adequate detail. This is what I can recall from reading the description.

stopWithLimitedRetry and startWithLimitedRetry add some additional FINAL states to terminate any operation that is in progress. This used to be 'FAILED' only. (So while STOPPING nodes, if a node goes into TERMINATED state - that is an acceptable outcome, and we should not loop till the timeout because this is not the intended 'stopped' state)

This is handled via the 'completedStatuses' set in the poller after an operation has started.
The stop/start APIs on AWS however don't behave very well if the instance being provided to them is in a final state of some kind. So if we invoke stop on an instance which is already 'TERMINATED' for some reason - the entire call fails.
The intent of the jira, IIRC, is to filter out such nodes before invoking the api ... so that it does not fail.

States per the AWS SDK
0 : pending
16 : running
32 : shutting-down
48 : terminated
64 : stopping
80 : stopped

The overall intent would be to avoid calling 'stop'/'start' if the instance is already in the 'shutting-down', 'terminated' states.


@Vatsal23082000  - conversion from CB modeled states to cloud-provider states may be required, or changing the invocation to include these states separately. The reverse translation is available
The original set of states in the start/stopWithLimitedRetry - likely don't make make sense for this initial filter. The only one really required would be the 'terminated', 'shutting-down' states.

Reply:
Done